Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -797,6 +825,61 @@ AXTextMarker AXTextMarker::findMarker(AXDirection direction, AXTextUnit textUnit
         }
         return linePosition;
     }
+
+    if (textUnit == AXTextUnit::Word) {
+        unsigned offset = this->offset();
+        AXTextMarker resultMarker = *this;
+
+        String flattenedRuns = currentRuns->substring(0, currentRuns->totalLength());
+
+        // objectBorder maintains the position in flattenedRuns between the current object's text and the previously scanned object(s)
+        int objectBorder = direction == AXDirection::Next ? 0 : flattenedRuns.length();
+        while (currentObject) {
+            int start, end;
+            findWordBoundary(flattenedRuns, offset, &start, &end);
+            if (direction == AXDirection::Previous) {
+                // allow previous word to start at text marker if we are in the middle of the word
+                // check that end is within the word using the wordLength
```

Comment:
WebKit coding style typically aims for "sentence cased" comments (Capital letter to start, period at the end).

Would it be more accurate to say something like:

"If `this` AXTextMarker points to the middle of a word, the previous word is..."

Not sure how to finish the sentence because I'm actually a little confused 🤔 I read your comment as saying the previous word would be this text marker? Is that right?

Reply:
This should read more like: "If this points to the middle of a word, the previous word begins at the start of that word"

Reply:
Took another look at this comment, and I don't think it makes sense anymore in the place it is at. Instead, there should be a comment above this line explaining this:
`if (start != (int)offset || end == (int)offset) {`